docs: explain page organization (#10476)
openfoodfacts/openfoodfacts-server: docs/index.md

The repository for the project is at https://github.com/openfoodfacts/openfoodfacts-server/

Do not hesitate to contribute to this documentation, this is highly appreciated.

> ## documentation organization

Reviewer: ```suggestion
## Documentation organization
```

> The documentation is splet between API documentation (for re-users, third party developpers) and Developper documentation (for contributors).

Reviewer: ```suggestion
The documentation is split between API documentation (for re-users, third party developers) and Developer documentation (for contributors).
```

The documentation follows the [Diátaxis framework](https://diataxis.fr/).

> Pages title should start with:

Reviewer: ```suggestion
Page titles should start with:
```
